### ISSUE TITLE
feature request: One liner installation in Windows

### ISSUE THREAD

**Issue by jackm9367** [logto-io/logto#1607]
### What problem did you meet?

The one-liner script outlined in README.md does not function on Windows due to the script utilising curl, printf, and other Linux/Bash commands. When run, the script immediately errors and halts. 

### Describe what you'd like Logto to have

A separate install script for Windows, including a separate Windows-oriented one liner that downloads and runs a wnidows-installer.js file in node, allowing for a one-line windows install.

**Comment by jackm9367** [comment]
Assign issue to self (@jackm9367)

**Comment by gao-sun** [comment]
thank you @jackm9367 - I didn't test developing in Windows, and it looks like the issue is straightforward. we can give it a try to see the effort. this issue can keep, no worries.

**Comment by demonzoo** [comment]
Would you like to try it in git bash env? I did it couple months ago and I remember most of the errors were gone, except for the husky after-install script.

**Comment by jackm9367** [comment]
Will give it a go now
I'd be guessing though that if Logto only 'supports' windows through `git bash`, then it doesn't really count as windows support?

**Comment by demonzoo** [comment]
Thank you, Jack. I'll also take a look at it this weekend.

**Comment by demonzoo** [comment]
> Will give it a go now
> I'd be guessing though that if Logto only 'supports' windows through `git bash`, then it doesn't really count as windows support?

I agree! But I also think that `git bash` might just be working as a baseline for windows users for the time being, until we work out a full compatible windows script.

**Comment by charIeszhao** [comment]
Hi Jack, how are you doing lately? It seems it's been awhile since we talked about this last time. And due to the inactivity of more than 3 weeks, we are going to take it back from you and see if there's anyone else who is also interested in.

And btw, we are now promoting a "Logto bounty hunter" event, in which you can complete bounties and claim rewards (real money 💰💰💰). And I think this ticket can be put on the bounty board as well. More details about this event can be found from [here](https://github.com/logto-io/logto/issues/1861).

Let me know if you want to claim it back and I can assign it back to you. Cheers!

**Comment by Olyno** [comment]
Hi 👋

I would like to be assigned to this issue if possible please 😄

**Comment by jackm9367** [comment]
Hey @charIeszhao and @Olyno 
I've been off sick for the last while so would love it if you could take this one @Olyno!
Sorry for not messaging you @charIeszhao, fell off my radar a bit.....
- Jack

**Comment by charIeszhao** [comment]
No problem. @jackm9367 Sorry to hear that, but I hope you are OK now.
Feel free to let us know if you would like to contribute something. You are always welcome here

**Comment by gao-sun** [comment]
quick update: thanks to @Olyno we're going to launch a cross-platform CLI soon. but I cloud not figure out the permission issue that happened in GitHub Actions Windows, you can see this [workflow runs](https://github.com/logto-io/logto/actions/workflows/integration-test.yml?page=1) down from #4640 - fought with Windows for a day but no luck.

I decided to move on as I'm a noob to Windows stuff meanwhile it is in CI which is hard to debug. I'll test on a real Windows laptop after launch tough. It'll be great if any of you guys would like to test the new CLI and help the CI issue.

**Comment by gao-sun** [comment]
guys, i'm wondering with the new CLI can Logto properly download and work in Windows?

**Comment by Olyno** [comment]
As the code is a JS code or a ``.exe`` executable, then yes it will work correctly on Windows.

**Comment by charIeszhao** [comment]
Guys, I just tried `npm init @logto` on windows, but I ended up only creating a `package.json` file...

**Comment by Olyno** [comment]
It actually makes sense. The ``npm init`` command (or ``yarn create`` or ``pnpm create`` depending on the package manager) works only if the ``name`` of the package starts with ``create-``.

For example, ``create-logto`` will make the package compatible with the command, but having a package named ``@logto`` will not work, because this one doesn't start with the ``create-`` prefix.

**Comment by charIeszhao** [comment]
Yes, but the same command `npm init @logto` works on macOS and Linux... So are you suggesting that we should rename the package name to `create-logto`?

**Comment by Olyno** [comment]
From what I see, it is mostly a bug on Mac and Linux. The package managers should not find the package in question according to their description:
 - https://docs.npmjs.com/cli/v8/commands/npm-init
 - https://classic.yarnpkg.com/lang/en/docs/cli/create/

With Yarn and Npm, they specify in their docs that the package name must start with the prefix "create-".

What I would suggest on my side is to rename the package to be conventional. This could fix the problem.

**Comment by charIeszhao** [comment]
Thanks, @Olyno. That makes sense to me.
@gao-sun Any thoughts?

**Comment by charIeszhao** [comment]
lol, again?? Let me check it real quick...

**Comment by gao-sun** [comment]
The "@" is for scoped package which follows another convention "@org/create". 

Actually I registered "create-logto" but didn't published anything yet (just a package.json). Since I'd like to publish it automatically instead of manually creating another package with the same content of "@logto/create".

**Comment by Olyno** [comment]
Nevermind. As Gao said, it is possible to use the organization as a prefix for the "npm init" command. However, this would make the package exclusive to npm, and other package managers would be excluded (Yarn, Pnpm, Deno and potentially Bun).

Even if I understand that creating a package with another name is annoying, this seems to me to be the best thing to do (and therefore deprecated "@logto/create").

You can take the example of [Redwood](https://github.com/redwoodjs/redwood) to see how they manage their packages.

**Comment by gao-sun** [comment]
thanks @Olyno, I'll take a look

**Comment by gao-sun** [comment]
@Olyno 
- I think both `npm create @logto` and `pnpm create @logto` are working fine
- for yarn, it just [hard-code the command](https://github.com/yarnpkg/yarn/blob/158d96dce95313d9a00218302631cd263877d164/src/cli/commands/create.js#L61) to run (you can dive to `coerceCreatePackageName()` if interested), and it will always be `create` for `yarn create @logto`. thus there's no way to use it. but yarn 1 is kind of archived. so i think it's ok to leave as it is.

for the traditional `create-logto` approach, rather than keeping it in sync of `@logto/create`, i think we can just print a error message and lead users to use `@logto/create` instead since all our packages are scoped.

how does it sound?

**Comment by Olyno** [comment]
The main problem is that the CLI does not work on Window. Creating a package under the organization (``@logto/create``) does not seem to fix this problem, quite the opposite.

I understand your concern about creating a package that doesn't contain the organization in the name, but I still think that this is the best way to fix the problem on Window, and make the package compatible under any manager package.

**Comment by gao-sun** [comment]
@Olyno thanks for the feedback and sorry for the delayed response. since i don't have a Windows dev environment, @charIeszhao would you like to have a try on this?

**Comment by wangsijie** [comment]
Thanks for reporting this issue.

Since then, the installation flow has been migrated from the Unix shell-based installer to a Node.js CLI. The current installer uses Node.js APIs for downloading and extracting Logto, so it no longer relies on commands such as `curl`, `printf`, or Unix-specific environment variable syntax.

The core issue should therefore be resolved, and Logto may now be installable on Windows through:

```bash
npm init @logto
```

or:

```bash
npm create @logto
```

Please give it another try on Windows. The current requirements are Node.js 22.14.0 or later and PostgreSQL 14 or later. We have not yet exhaustively verified every combination of Windows shell and package manager, so if it still fails, please share the command, shell, Node.js version, package manager, and error output. That will help us identify any remaining Windows-specific issues.
